neo4j/v1/session.py

```diff
@@ -477,6 +478,12 @@ def healthy(self):
         return None if connection.closed else not connection.defunct
 
     def run(self, statement, parameters=None):
+        if self.transaction:
+            raise ProtocolError("Please close the currently open transaction object before running more "
+                                "statements/transactions in the current session.")
```

Comment:
Instead of suggesting the transaction be closed as the only course of action, we should say something like:

```
Statements cannot be run directly on a session with an open transaction; either run from within the transaction or use a different session.
```